lib/modules/manager/helmfile/artifacts.ts

```diff
@@ -59,7 +69,44 @@ export async function updateArtifacts({
         constraint: config.constraints?.kustomize,
       });
     }
-    await exec(`helmfile deps -f ${quote(packageFileName)}`, {
+
+    const cmd: string[] = [];
+    // set cache and config files to a path in privateCacheDir to prevent file and credential leakage
+    const helmConfigParameters = [
+      `--registry-config ${upath.join(privateCacheDir(), 'registry.json')}`,
+      `--repository-config ${upath.join(privateCacheDir(), 'repositories.yaml')}`,
+      `--repository-cache ${upath.join(privateCacheDir(), 'repositories')}`,
+    ];
+    const doc = yaml.load(newPackageFileContent) as Doc; //TODO #9610
+    const ociRepositoryRules: RepositoryRule[] = getRepositories(doc)
+      .filter(isOCIRegistry)
+      .map((value) => {
+        return {
+          ...value,
+          hostRule: hostRules.find({
+            url: 'https://' + value.url,
+            hostType: DockerDatasource.id,
+          }),
+        };
+    });
+
+    const regexOfURLPath = /\/.*/;
+    ociRepositoryRules.forEach((value) => {
+      const { username, password } = value.hostRule;
+      const parameters = [...helmConfigParameters];
+      if (username && password) {
+        parameters.push(`--username ${quote(username)}`);
+        parameters.push(`--password ${quote(password)}`);
+        const host = value.url.replace(regexOfURLPath, '')
+
+        cmd.push(
+          `helm registry login ${parameters.join(' ')} ${host}`
+        );
+      }
+    });
+
+    cmd.push(`helmfile deps -f ${quote(packageFileName)}`)
```

Comment:
don't you need to add the config parameter here too?

Reply:
ohh, i forgot.